content/client-lib-development-guide/features.textile

```diff
@@ -1473,6 +1495,82 @@ enum StatsIntervalGranularity:
   DAY
   MONTH
 
+class DeviceDetails:
+  + fromLocalDevice() => io DeviceDetails
+  id: String
+  platform: DevicePlatform
+  formFactor: DeviceFormFactor
+  clientId: String?
+  metadata: Dict<String, String>
+  updateTokenId: String
+  resetId(String)
```

Comment:
I thought we had agreed on just `resetId` i.e. not allowing a user to explicitly set the device ID?  Given the registration target is linked to a unique device ID, I am not sure I can think of  a good reason when a dev should be able to set the device ID.